### DISCUSSION TITLE
Erroneous shell variable parsing with jq

### DISCUSSION THREAD

**Discussion by pantelis-karamolegkos** [community/community#37802] [Actions] [answered]
I have the following GitHub Actions step

```
        run: |
          COMMENT_BODY=`echo $GITHUB_CONTEXT | jq '.event.comment.body'`
          COMMENT_USER=`echo $GITHUB_CONTEXT | jq '.event.comment.user.login'`
          MEMBERS=`gh api -H "Accept: application/vnd.github+json" /orgs/MyOrg/teams/${{ env.ALLOWEED_OVERRIDING_TEAM_NAME }}/members`
          TESTING=`echo $MEMBERS | jq 'any(.login == "\"$COMMENT_USER\"")'`
          echo $TESTING
```

For some reason, `TESTING` evaluates to `false`.

However, when I hardcode my username as follows:

```
TESTING=`echo $MEMBERS | jq 'any(.login == "pkaramol")'`
```

it evaluates to `true`.

Is there something wrong with my substitution in the `jq` invocation?

(I know `$COMMENT_USER` evaluates to `pkaramol` by the action logs)

**Comment by airtower-luna** [comment, marked as answer]
The problem are the single quotes, the shell doesn't do variable substitution between those. I think what you want is something like:

```
TESTING=`echo $MEMBERS | jq "any(.login == \"${COMMENT_USER}\")"`
```
